src/mlpack/tests/lsh_test.cpp

```diff
@@ -752,4 +752,79 @@ BOOST_AUTO_TEST_CASE(EmptyConstructorTest)
   BOOST_REQUIRE_EQUAL(distances.n_rows, 3);
 }
 
+// These two tests are only compiled if the user has specified OpenMP to be
+// used.
+#ifdef HAS_OPENMP
+/**
+ * Test: This test verifies that parallel query processing returns correct
+ * results for the bichromatic search.
+ */
+BOOST_AUTO_TEST_CASE(ParallelBichromatic)
+{
+  // kNN and LSH parameters (use LSH default parameters).
+  const int k = 4;
+  const int numTables = 16;
+  const int numProj = 3;
+
+  // Read iris training and testing data as reference and query sets.
+  const string trainSet = "iris_train.csv";
+  const string testSet = "iris_test.csv";
+  arma::mat rdata;
+  arma::mat qdata;
+  data::Load(trainSet, rdata, true);
+  data::Load(testSet, qdata, true);
+
+  // Where to store neighbors and distances
+  arma::Mat<size_t> sequentialNeighbors;
+  arma::Mat<size_t> parallelNeighbors;
+  arma::mat distances;
+
+  // Construct an LSH object. By default, it uses the maximum number of threads
+  LSHSearch<> lshTest(rdata, numProj, numTables); //default parameters
+  lshTest.Search(qdata, k, parallelNeighbors, distances);
+
+  // Now perform same search but with 1 thread
+  lshTest.MaxThreads(1);
```

Comment:
If you did remove the `MaxThreads()` member from `LSHSearch`, you could easily replicate the same functionality by just calling `omp_set_num_threads(1)` before the second call to `lshTest.Search()`.  (Maybe you would want to restore the original number of threads afterwards for other OpenMP tests.)